packages/backend/src/modules/tvs/mapConfig.ts

```diff
@@ -29,77 +40,128 @@ export function mapConfig(project: BackendProject, chain: string): TvsConfig {
-  const amountFormula = token.amount as BalanceOfEscrowAmountFormula
-
-  // add escrow to the amount formula
-  amountFormula.escrowAddresses.push(escrow.address)
+function updateEscrowToken(
+  token: EscrowToken,
+  escrow: BackendProjectEscrow,
+): Token {
+  // add this escrow to tokens esccrows list
+  const escrowAddresses = [...token.amount.escrowAddresses, escrow.address]
 
+  let sinceTimestamp = token.sinceTimestamp
   // update sinceTimestamp if needed
-  if (escrow.sinceTimestamp < token.sinceTimestamp) {
-    token.sinceTimestamp = escrow.sinceTimestamp
+  if (escrow.sinceTimestamp.lt(token.sinceTimestamp)) {
+    sinceTimestamp = escrow.sinceTimestamp
   }
 
+  let untilTimestamp = token.untilTimestamp
   // reset or update  untilTimestamp if needed
   if (!escrow.untilTimestamp && token.untilTimestamp) {
-    token.untilTimestamp = undefined
+    untilTimestamp = undefined
   } else if (
     escrow.untilTimestamp &&
     token.untilTimestamp &&
-    escrow.untilTimestamp > token.untilTimestamp
+    escrow.untilTimestamp.gt(token.untilTimestamp)
   ) {
-    token.untilTimestamp = escrow.untilTimestamp
+    untilTimestamp = escrow.untilTimestamp
+  }
+
+  return {
+    ...token,
+    amount: {
+      ...token.amount,
+      escrowAddresses,
+    },
+    sinceTimestamp,
+    untilTimestamp,
   }
 }
 
 function createToken(
   legacyToken: LegacyToken,
-  escrow: BackendProjectEscrow,
   project: BackendProject,
-  chain: string,
+  chain: ChainConfig,
+  escrow?: BackendProjectEscrow,
 ): Token {
   let amountFormula: AmountFormula
+  let sinceTimestamp: UnixTime
+  let untilTimestamp: UnixTime | undefined
 
   switch (legacyToken.supply) {
     case 'zero':
+      assert(escrow, 'Escrow is required for zero supply tokens')
+
       amountFormula = {
         type: 'balanceOfEscrow',
         address: legacyToken.address ?? EthereumAddress.ZERO,
-        chain,
+        chain: chain.name,
         escrowAddresses: [escrow.address],
         decimals: legacyToken.decimals,
       } as BalanceOfEscrowAmountFormula
+
+      sinceTimestamp = escrow.sinceTimestamp
+      untilTimestamp = escrow.untilTimestamp
       break
     case 'totalSupply':
+      assert(
+        chain.minTimestampForTvl,
+        'Chain with token should have minTimestamp',
+      )
+
       amountFormula = {
         type: 'totalSupply',
         address: legacyToken.address,
-        chain,
+        chain: chain.name,
         decimals: legacyToken.decimals,
       } as TotalSupplyAmountFormula
+
+      sinceTimestamp = UnixTime.max(
+        chain.minTimestampForTvl,
+        legacyToken.sinceTimestamp,
+      )
       break
     case 'circulatingSupply':
+      assert(
+        chain.minTimestampForTvl,
+        'Chain with token should have minTimestamp',
+      )
+
       amountFormula = {
         type: 'circulatingSupply',
         ticker: legacyToken.symbol,
       } as CirculatingSupplyAmountFormula
+
+      sinceTimestamp = UnixTime.max(
```

Comment:
IMO this should be fixed "higher" in the abstraction chain, token should not be allowed to be earlier than the chain. 
I think somewhere before this mapConfig function we will have function e.g. mergeConfigs(chains,tokens) which will handle this one.
I propose we emulate this here already, so we do not forget in the future. Because setting those min and max timestamps for configs on chain is one of the biggest pinpoints of current infra